daemon/daemon_test.go

```diff
@@ -335,6 +337,54 @@ func TestDaemonReloadLabels(t *testing.T) {
 	}
 }
 
+func TestDaemonReloadInsecureRegistries(t *testing.T) {
+	daemon := &Daemon{}
+	daemon.RegistryService = registry.NewService(registry.ServiceOptions{})
+	daemon.configStore = &Config{}
+
+	valuesSets := make(map[string]interface{})
+	valuesSets["insecure-registries"] = "10.10.1.11:5000"
+
+	insecureRegistries := []string{"127.0.0.0/8", "10.10.1.11:5000", "10.10.1.12:5000"}
+	newConfig := &Config{
+		CommonConfig: CommonConfig{
+			ServiceOptions: registry.ServiceOptions{
+				InsecureRegistries: insecureRegistries,
+			},
+			valuesSet: valuesSets,
+		},
+	}
+
+	daemon.Reload(newConfig)
+
+	registries := daemon.RegistryService.ServiceConfig()
+
+	dataMap := map[string]int{}
+	for _, value := range registries.InsecureRegistryCIDRs {
```

Comment:
This is really confusing to me (from here to the end of the test).
Is there a way to clean this up and check explicit values?